src/pytest_inmanta_lsm/lsm_project.py

```diff
@@ -462,48 +845,100 @@ def compile(
             self.project.compile(model, no_dedent=False)
             return
 
-        if str(service_id) not in self.services:
-            raise ValueError(
-                f"Can not find any service with id {service_id} in our inventory.  "
-                "Did you add it using the add_service method?"
-            )
+        # Get the service targeted by the compile
+        service = self.get_service(service_id)
 
-        service = self.services[str(service_id)]
+        env: dict[str, str] = {}
+        env[inmanta_lsm.const.ENV_INSTANCE_ID] = str(service_id)
+        env[inmanta_lsm.const.ENV_INSTANCE_VERSION] = str(service.version)
+
+        try:
+            env[inmanta_lsm.const.ENV_PARTIAL_COMPILE] = str(self.partial_compile)
+        except AttributeError:
+            # This attribute only exists for iso5+, iso4 doesn't support partial compile.
+            # We then simply don't set the value.
+            if self.partial_compile:
+                warnings.warn("Partial compile is enabled but it is not supported, it will be ignored.")
+
+        if validation:
+            # If we have a validation compile, we need to set an additional env var
+            env[inmanta_lsm.const.ENV_MODEL_STATE] = inmanta_lsm.model.ModelState.candidate
+
+        LOGGER.debug(
+            "Triggering compile for service %s (%s) with the following environment variables: %s",
+            service.id,
+            service.service_entity,
+            env,
+        )
+        with self.monkeypatch.context() as m:
+            for k, v in env.items():
+                m.setenv(k, v)
+            self.project.compile(model, no_dedent=False)
+
+    def post_partial_compile_validation(
```

Comment:
It would be nice to have a test of this feature too.

Reply:
Good point, I added it